lib/rucio/core/transfer.py

```diff
@@ -750,40 +857,21 @@ def get_transfer_requests_and_source_replicas(total_workers=0, worker_number=0,
                 reqs_scheme_mismatch.append(req_id)
             continue
 
-        # This loop is to fill the rses_info and rse_mapping dictionary for the intermediate RSEs including the dest_rse_id
-        for hop in list_hops:
-            # hop = {'source_rse_id': source_rse_id, 'source_scheme': 'srm', 'dest_rse_id': dest_rse_id, 'dest_scheme': 'srm'}
-            if hop['dest_rse_id'] not in rse_mapping:
-                rse_mapping[hop['dest_rse_id']] = get_rse_name(rse_id=hop['dest_rse_id'], session=session)
-            if hop['dest_rse_id'] not in rses_info:
-                rses_info[hop['dest_rse_id']] = rsemgr.get_rse_info(rse=rse_mapping[hop['dest_rse_id']],
-                                                                    vo=get_rse_vo(rse_id=hop['dest_rse_id'], session=session),
-                                                                    session=session)
-            if hop['dest_rse_id'] not in rse_attrs:
-                rse_attrs[hop['dest_rse_id']] = get_rse_attributes(hop['dest_rse_id'], session=session)
-
-        source_protocol = list_hops[0]['source_scheme']
-        destination_protocol = list_hops[-1]['dest_scheme']
+        source_scheme = list_hops[0]['source_scheme']
+        dest_scheme = list_hops[-1]['dest_scheme']
         dest_scheme_priority = list_hops[-1]['dest_scheme_priority']
 
-        transfer_src_type = "DISK"
-        transfer_dst_type = "DISK"
         allow_tape_source = True
         try:
-            if rses and dest_rse_id not in rses:
-                continue
-
-            # Get the source rse information
-            if source_rse_id not in rses_info:
-                rses_info[source_rse_id] = rsemgr.get_rse_info(rse=source_rse_name,
-                                                               vo=get_rse_vo(rse_id=source_rse_id, session=session),
-                                                               session=session)
-            if source_rse_id not in rse_attrs:
-                rse_attrs[source_rse_id] = get_rse_attributes(source_rse_id, session=session)
             # Get source protocol
-            source_rse_id_key = 'read_%s_%s' % (source_rse_id, source_protocol)
-            if source_rse_id_key not in protocols:
-                protocols[source_rse_id_key] = rsemgr.create_protocol(rses_info[source_rse_id], 'third_party_copy', source_protocol)
+            source_protocol = ctx.protocol(source_rse_id, source_scheme, 'read')
```

Comment:
Also changes the current behavior. Probably correct idea, but must be fixed separately. Otherwise we introduce new features in a PR which is about refactoring. If problems due to this PR are discovered, it will be difficult to distinguish between it being caused by the refactoring; or by the new feature.